packages/ui5-proxy-middleware/src/base/utils.ts

```diff
@@ -0,0 +1,302 @@
+            */
+            proxyRes.headers['Etag'] = etag;
+            proxyRes.headers['cache-control'] = 'no-cache';
+            resolve(responseBuffer);
+        }
+    });
+};
+
+/**
+ * Handler for the proxy request event.
+ * Re-validates the cached UI5 sources based on the ETag.
+ * Logs the requests made by the proxy.
+ *
+ * @param proxyReq - proxy request object
+ * @param res - server response object
+ * @param etag - Etag of the cached UI5 sources, normally the UI5 version
+ * @param logger - Logger for loging the requests
+ */
+export const proxyRequestHandler = (
+    proxyReq: ClientRequest,
+    res: ServerResponse,
+    etag: string,
+    logger: ToolsLogger
+): void => {
+    logger.info(proxyReq.path);
+    if (proxyReq.getHeader('if-none-match') === etag) {
+        res.statusCode = 304;
+        res.end();
+    }
+};
+
+/**
+ * Get user's proxy configuration.
+ *
+ * @param yamlProxyServer - proxy server config from yaml file
+ * @returns User's proxy configuration or undefined
+ */
+export const getCorporateProxyServer = (yamlProxyServer: string | undefined): string | undefined => {
+    return (
+        yamlProxyServer ||
+        process.env.FIORI_TOOLS_PROXY ||
+        process.env.http_proxy ||
+        process.env.HTTP_PROXY ||
+        process.env.https_proxy ||
+        process.env.HTTPS_PROXY ||
+        process.env.npm_config_proxy ||
+        process.env.npm_config_https_proxy
+    );
+};
+
+/**
+ * Checks if a host is excluded from user's corporate proxy.
+ *
+ * @param noProxyConfig - user's no_proxy configuration
+ * @param host - host to be checked
+ * @returns true if host is excluded from user's corporate server, false otherwise
+ */
+export const isHostExcludedFromProxy = (noProxyConfig: string | undefined, host: string): boolean => {
+    const noProxyList = noProxyConfig ? noProxyConfig.split(',') : [];
+    const found = noProxyList.find((entry) => {
+        return host.includes(entry);
+    });
+    return !!found;
+};
+
+/**
+ * Returns the name of html file, which is used to preview the application, from the URL.
+ *
+ * @param url - html request url
+ * @returns Name of the html file
+ */
+export const getHtmlFile = (url: string): string => {
+    let html = url;
+    if (html.indexOf('?') !== -1) {
+        html = html.split('?')[0].replace(/["']/g, '');
+    } else if (html.indexOf('#') !== -1) {
+        html = html.split('#')[0].replace(/["']/g, '');
+    } else {
+        html = html.replace(/["']/g, '');
+    }
+
+    return html;
+};
+
+/**
+ * Returns the name of the yaml file, which is used to for the server configuration, from the runtime arguments.
+ *
+ * @param args - runtime arguments
+ * @returns Name of the YAML file
+ */
+export const getYamlFile = (args: string[]): string => {
+    let yaml = 'ui5.yaml';
+    const index = args.indexOf('--config') !== -1 ? args.indexOf('--config') : args.indexOf('-c');
+
+    if (index !== -1) {
+        yaml = args[index + 1];
```

Comment:
This seems to assume that `-c some-file.yaml` or `--config some-file.yaml` is _always_ the last argument. This makes the CLI a bit brittle, can we think of some other way of handling CLI arguments?

Reply:
This searches for the existence of `-c` or `--config` in the list of arguments. If one of those exists, then the next entry in the list is the value for `-c` or `--config`. Why do you think that this is always the last argument?

Reply:
> Why do you think that this is always the last argume
I misread the code then, sorry.

How about the case when the config file name is not passed in?

Reply:
When the config file name is not passed in, then `ui5.yaml` is taken as default config name. Same behaviour like the `ui5/cli`.